src/arrayMethodSplice.js

```diff
@@ -5,7 +5,188 @@
  */
 function applyCustomSplice() {
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    const removedValues = [];
+    let startIndex = start;
+    const howManyDelete = deleteCount;
+    const addItems = items;
+
+    // setting the start point
+    if (startIndex > this.length) {
+      startIndex = this.length;
+    }
+
+    // if first parameter is undefined
+    // or a negative index greater than the length of the array
+    if (startIndex === undefined || startIndex < (this.length * -1)) {
+      startIndex = 0;
+    } else if (startIndex < 0) {
+      startIndex += this.length;
+    }
```

Comment:
Better to move logic of startIndex normalisation to separate function, your function is too big, better to split logic